src/electionguard/manifest.py

```diff
@@ -805,6 +805,53 @@ def is_valid(self) -> bool:
             )
         return success
 
+    def _get_candidate_name(self, candidate: Candidate, lang: str) -> str:
+        if candidate.is_write_in:
+            return "Write-In"
+        query = (t.value for t in candidate.name.text if t.language == lang)
+        name = next(query, None)
+        name = candidate.object_id if name is None else name
+        return name
+
+    def _get_candidate_names(self, lang: str) -> Dict[str, str]:
+        return {
+            candidate.object_id: self._get_candidate_name(candidate, lang)
+            for candidate in self.candidates
+        }
+
+    def _get_selections_with_candidate_id(self) -> Dict[str, str]:
+        selections: Dict[str, str] = {}
+        for contest in self.contests:
+            for selection in contest.ballot_selections:
+                selections.update({selection.object_id: selection.candidate_id})
+        return selections
+
+    def _replace_candidate_ids_with_names(
+        self, selections: Dict[str, str], candidates: Dict[str, str]
+    ) -> None:
+        for (selection_id, candidate_id) in selections.items():
+            candidate_name = candidates.get(candidate_id)
+            if candidate_name is not None:
+                selections.update({selection_id: candidate_name})
+
+    def get_selection_names(self, lang: str) -> Dict[str, str]:
+        """
+        Retrieves a dictionary whose keys are all selection id's and whose values are
+        those selection's candidate names in the supplied language if available
+        """
+        candidates = self._get_candidate_names(lang)
+        selections = self._get_selections_with_candidate_id()
+        self._replace_candidate_ids_with_names(selections, candidates)
+        return selections
+
+    def get_contest_names(self) -> Dict[str, str]:
```

Comment:
These are pretty specific and not a requirement of something that should be a `dataclass`. I would move these to functional methods or a wrapping class both of which could exist in `electionguard_tools/helpers` or the `cli` directly.

`get_contest_names(manifest: Manifest)`

or 

`class PrintableManifest (Manifest)`